### PR TITLE
FPO-290

### PR DESCRIPTION
**Pull request by Lee365**
### Jira link
https://transformuk.atlassian.net/browse/FPO-290

### What?

I have added/removed/altered:

- [x] Added
- [x] Removed
- [x] Altered

### Why?

I am doing this because:

-

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: src/services/organisationService.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          organisationId
+        }
+      )
+    } catch (error) {
+      logger.error('Error fetching Organisation', error)
+      throw error
+    }
+  }
+
+  export async function updateOrganisation (organisationId: string, applicationReference: string): Promise<Organisation> {
```
Comment (ashward) [inline src/services/organisationService.ts:35]: `updateOrganisation` implies something generic, but this is very specifically just updating the reference number and resetting the status. Maybe give this function a more specific name according to what it's actually doing?

File: src/controllers/verificationController.ts [hunk 2]
```diff
@@ -102,13 +102,13 @@ export const applicationComplete = async (req: Request, res: Response): Promise<
...
       const env = process.env.NODE_ENV ?? 'development'
+      const referenceNumber: string = generateApplicationReference(8)
       if (env === 'production') {
-        const referenceNumber: string = generateApplicationReference(8)
         await EmailService.sendEmail(registrationTemplateId, session.emailAddress as string, session.organisationName as string, referenceNumber)
         await EmailService.sendEmail(applicationTemplateId, applicationSupporteEmail, session.organisationName as string, referenceNumber)
       }
       req.session = null
-      res.render('completion')
+      res.render('completion', { referenceNumber })
```
Comment (ashward) [inline src/controllers/verificationController.ts:111]: This is called `applicationReference` elsewhere. Might be good to keep it consistent.

File: src/services/organisationService.ts (hunk 1 above)
Comment (Lee365) [reply src/services/organisationService.ts:35]: updateOrganisationStatusAndReference

File: src/controllers/verificationController.ts (hunk 2 above)
Comment (Lee365) [reply src/controllers/verificationController.ts:111]: Changed to applicationReference

**Review by mkayad — APPROVED** [review]
LGTM

**Pull request merged**